pkg/controller/lifecycle/actuator.go

```diff
@@ -472,33 +475,71 @@ func getSeedResources(oidcReplicas *int32, hibernated bool, namespace, genericKu
 	}
 
 	if oidcReplicas != nil && *oidcReplicas > 0 {
-		err = registry.Add(&autoscalingv2beta1.HorizontalPodAutoscaler{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      constants.ApplicationName,
-				Namespace: namespace,
-			},
-			Spec: autoscalingv2beta1.HorizontalPodAutoscalerSpec{
-				ScaleTargetRef: autoscalingv2beta1.CrossVersionObjectReference{
-					APIVersion: appsv1.SchemeGroupVersion.String(),
-					Kind:       "Deployment",
-					Name:       constants.ApplicationName,
+		var k8sVersionGreaterOrEqualThan123 bool
```

Comment:
You can use a check similar to [this one](https://github.com/gardener/gardener/blob/1da39002ccd90e8b3dc08763dcf41708e48d907f/pkg/operation/botanist/component/nodeproblemdetector/node_problem_detector.go#L404) in order to match a desired k8s version.
Also do not overwrite the whole security context, but only add the seccomp profile (see the above example).

You can extract the whole logic for the hpa resource to a different function so it is more readable.